Include all active sessions in the web API

zmb3 opened the pull request:
Depends on gravitational/webapps#1315

Updates #17871

zmb3 committed:
Include active desktop sessions in the web API

Updates #17126

zmb3 committed:
Include all active sessions in API response

Also update the active sessions test to operate based on the presence
of session trackers rather than actually starting a session. This
allows us to test additional session types and is more deterministic.

Updates #17871

kimlisa commented on api/types/session_tracker.go:
```diff
@@ -99,6 +99,15 @@ type SessionTracker interface {
 	GetKubeCluster() string
 
+	// GetDesktopName returns the name of the Windows desktop the session is running in.
+	GetDesktopName() string
```
weird, i don't see this being a duplicate as `github actions/analyze` says it is...

kimlisa commented on lib/web/apiserver_test.go:
```diff
@@ -1406,46 +1406,80 @@ func TestWebAgentForward(t *testing.T) {
-	t.Cleanup(func() { require.NoError(t, ws.Close()) })
+	start := time.Now()
+	kinds := []types.SessionKind{
+		types.SSHSessionKind,
```
add `KubernetesSessionKind` while at it?

zmb3 committed:
Correct the name for app/db sessions

For app sessions, use the name from the types.Application, since
the value in the cert is only present for CLI sessions.

For database sessions, use the database service name, which maps
to the Teleport database service the user is connecting to, since
the actual "logical" database name is not required.

greedy52 commented on lib/web/apiserver.go:
```diff
@@ -2209,6 +2210,9 @@ func trackerToLegacySession(tracker types.SessionTracker, clusterName string) se
 		ServerAddr:            tracker.GetAddress(),
 		ClusterName:           clusterName,
 		KubernetesClusterName: tracker.GetKubeCluster(),
+		DesktopName:           tracker.GetDesktopName(),
+		AppName:               tracker.GetAppName(),
+		DatabaseName:          tracker.GetDesktopName(),
```
should this be `GetDatabaseName`?

zmb3 replied:
🤦‍♂️

zmb3 committed:
Merge branch 'master' into zmb3/active-desktop-sessions
